app/javascript/app/pages/ndcs/ndcs-reducers.js

```diff
@@ -18,6 +18,7 @@ export default {
           ...state,
           data: {
             categories: payload.categories,
+            mapCategories: state.data.mapCategories || {},
```

Comment:
Is needed to generate a new array / object on the reducer? 

I mean, this will trigger a new render as it is a new instance so why not extend the data using something like?
```
data: {
  ...state.data,
  theNewParamsThatWeHave...
}
```

and update only the params that we need

Reply:
+1